api.py

```diff
@@ -230,14 +235,41 @@ def cs_terms_query(q: str, field: str = "article_title", aggr: str = "top"):
     return query
 
 
-def cs_paged_query(q: str, resume: Union[str, None] = None, expanded: bool = False):
+def _validate_sort_order(sort_order: Optional[str]):
+    if sort_order and sort_order not in VALID_SORT_ORDERS:
+        raise HTTPException(status_code=400,
+                            detail=f"Invalid sort order (must be on of {', '.join(VALID_SORT_ORDERS)})")
+
+
+def _validate_sort_field(sort_field: Optional[str]):
+    if sort_field and sort_field not in VALID_SORT_FIELDS:
+        raise HTTPException(status_code=400,
+                            detail=f"Invalid sort field (must be on of {', '.join(VALID_SORT_FIELDS)})")
+
+
+def _validate_page_size(page_size: Optional[int]):
+    if page_size and page_size < 1:
+        raise HTTPException(status_code=400,
+                            detail=f"Invalid page size (must be greater than 0)")
+
+
+def cs_paged_query(q: str, resume: Optional[str], expanded: Optional[bool], sort_field=Optional[str],
+                   sort_order=Optional[str], page_size=Optional[int]) -> Dict:
     query = cs_basic_query(q, expanded)
+    final_sort_field = sort_field or "publication_date"
+    _validate_sort_field(final_sort_field)
+    final_sort_order = sort_order or "desc"
+    _validate_sort_order(final_sort_order)
+    final_page_size = page_size or config["maxpage"]
+    _validate_page_size(final_page_size)
```

Comment:
If the validate_.... functions returned the validated value, the above six lines could be more compact:
`thing = _validate_thing(thing or "default_thing_value")`